backend/apps/chats/tests/test_chats_module.py

```
@@ -0,0 +1,812 @@
+        self.assertFalse(is_admin.has_permission(request_no_residence, SimpleNamespace()))
+        self.assertFalse(is_manager.has_permission(request_no_residence, SimpleNamespace(action="list")))
+        self.assertFalse(is_resident.has_permission(request_no_residence, SimpleNamespace()))
+
+        request_student = SimpleNamespace(user=self.student_user, residence=self.residence)
+        self.assertFalse(is_manager.has_permission(request_student, SimpleNamespace(action="create")))
+        self.assertTrue(is_manager.has_permission(request_student, SimpleNamespace(action="list")))
+
+    def test_serializers_unit_branches(self):
+        label_serializer = ChatGroupLabelSerializer(context={})
+        self.assertEqual(label_serializer.validate_name("  social  "), "social")
+        with self.assertRaises(Exception):
```

Ruff (0.15.6)

[warning] Do not assert blind exception: `Exception`

(B017)